lib/greengrass/discoveryclient.ts

```diff
@@ -10,7 +10,7 @@
  */
 
 import { io, http, CrtError } from 'aws-crt';
```

Comment:
These are native imports, for browser it should use `import { io, http, CrtError } from 'aws-crt/dist.browser/browser';`
The other clients are likely native only, breaking browser usage of the aws-iot-device-sdk-v2.

Reply:
Not necessarily.  For example, I don't have any problems running node's shadow sample in the browser given a credentials sourcing refactor.  The key to that though is that I'm using the webpack/tsconfigs from the browser samples which manage the switch under the scenes during the build process.

The intent of the CRT is to be polymorphic (as close as we can get it) across node and browser.  Unfortunately, it does not live up to that intent in its current form.  Currently, your install target/platform comes from config details in webpack and potentially tsconfig (not sure) which is not a good (clear, easy, works out-of-the-box) experience.

Clearly, the change broke you, but I think the actual problem is not the change itself, but the deeper problem of making browser/node CRT polymorphism invisible and painless.  

I'm not sure what the right path forward here is yet.